packages/swing-store/src/sqlStreamStore.js

```diff
@@ -54,6 +54,19 @@ export function sqlStreamStore(dbDir, io) {
     )
   `);
 
+  // We use explicit transactions to 1: not commit writes until the
+  // host application calls commit() and 2: avoid expensive fsyncs
+  // until the appropriate commit point. All API methods should call
+  // this first, otherwise sqlite will automatically start a
+  // transaction for us, but it will commit/fsync at the end of the DB
+  // run(). https://sqlite.org/lang_transaction.html
+  function maybeStartTransaction() {
```

Comment:
"maybe" suggests "might fail", not "might be already done", to me. Consider "ensure"